src/iuse_actor.cpp

```diff
@@ -719,3 +720,201 @@ long reveal_map_actor::use( player *p, item *it, bool, point ) const
+    }
+    if( g->m.get_field( pos, fd_fire ) ) {
+        // check if there's already a fire
+        p->add_msg_if_player(m_info, _("There is already a fire."));
+        return false;
+    }
+    if( g->m.flammable_items_at(pos.x, pos.y) ||
+        g->m.has_flag("FLAMMABLE", pos.x, pos.y) || g->m.has_flag("FLAMMABLE_ASH", pos.x, pos.y) ||
+        g->m.get_field_strength( pos, fd_web ) > 0 ) {
+        return true;
+    } else {
+        p->add_msg_if_player(m_info, _("There's nothing to light there."));
+        return false;
+    }
+}
+
+void firestarter_actor::resolve_firestarter_use( const player *p, const item *, const point &pos )
+{
+    if( g->m.add_field( pos, fd_fire, 1, 100 ) ) {
+        p->add_msg_if_player(_("You successfully light a fire."));
+    }
+}
+
+// TODO: Move prep_firestarter_use here
+long firestarter_actor::use( player *p, item *it, bool t, point pos ) const
+{
+    if (it->has_flag("REFILLABLE_LIGHTER")) {
+        if( p->is_underwater() ) {
+            p->add_msg_if_player(_("The lighter is extinguished."));
+            it->make("ref_lighter");
+            it->active = false;
+            return 0;
+        }
+        if (t) {
+            if (it->charges < it->type->charges_to_use()) {
+                p->add_msg_if_player(_("The lighter burns out."));
+                it->make("ref_lighter");
+                it->active = false;
+            }
+        } else if (it->charges <= 0) {
+            p->add_msg_if_player(_("The %s winks out."), it->tname().c_str());
+        } else { // Turning it off
+            int choice = menu(true, _("refillable lighter (lit)"), _("extinguish"),
+                              _("light something"), _("cancel"), NULL);
+            switch (choice) {
+                case 1: {
+                    p->add_msg_if_player(_("You extinguish the lighter."));
+                    it->make("ref_lighter");
+                    it->active = false;
+                    return 0;
+                }
+                break;
+                case 2:
+                    if( prep_firestarter_use(p, it, pos) ) {
+                        p->moves -= moves_cost;
+                        resolve_firestarter_use(p, it, pos);
+                        return it->type->charges_to_use();
+                    }
+            }
+        }
+        return it->type->charges_to_use();
+    } else { // common ligher or matches
+        if( prep_firestarter_use(p, it, pos) ) {
+            p->moves -= moves_cost;
+            resolve_firestarter_use( p, it, pos );
+            return it->type->charges_to_use();
+        }
+    }
+    return 0;
+}
+
+bool firestarter_actor::can_use( const player* p, const item*, bool, const point& ) const
+{
+    if( p->is_underwater() ) {
+        return false;
+    }
+
+    return true;
+}
+
+extended_firestarter_actor::~extended_firestarter_actor() {}
+
+void extended_firestarter_actor::load( JsonObject &obj )
+{
+    need_sunlight = obj.get_bool( "need_sunlight", false );
```

Comment:
You should initialize the parent whether you use its members or not, someone will try to use that member in the future and have to figure out why it's not being initialized.